src/layers/msgpass.jl

```diff
@@ -34,14 +34,18 @@ update_vertex(m::T; kwargs...) where {T<:MessagePassing} = update(m; kwargs...)
 aggregate_neighbors(m::T, aggr::Symbol; kwargs...) where {T<:MessagePassing} =
     pool(aggr, kwargs[:cluster], kwargs[:M])
 
+aggregate_neighbors(m::T, aggr::Symbol, M, cluster; kwargs...) where {T<:MessagePassing} =
+    pool(aggr, cluster, M)
+
 function propagate(mp::T; aggr::Symbol=:add, adjl=adjlist(mp), kwargs...) where {T<:MessagePassing}
     gi = GraphInfo(adjl)
 
     # message function
     M = update_edge(mp; gi=gi, kwargs...)
 
     # aggregate function
-    M = aggregate_neighbors(mp, aggr; M=M, cluster=generate_cluster(M, gi))
+    cluster = generate_cluster(M, gi)
```

Comment:
This is good. I will take this and avoid calculating adjoint on `generate_cluster`, because it is totally not differentiable.

Reply:
Indeed!